addons/binding/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/dpt/KNXCoreTypeMapper.java

```diff
@@ -0,0 +1,957 @@
+
+        /**
+         * MainType: 29
+         * 29.012: Reactive energy, values -9223372036854775808 9223372036854775807 VARh
+         * 29.011: Apparent energy, values -9223372036854775808 9223372036854775807 VAh
+         * 29.010: Active Energy, values -9223372036854775808 9223372036854775807 Wh
+         */
+        dptMainTypeMap.put(29, DecimalType.class);
+        /** Exceptions Datapoint Types "64-Bit Signed Value", Main number 29 */
+        // Example: dptTypeMap.put(DPTXlator64BitSigned.DPT_ACTIVE_ENERGY.getID(), DecimalType.class);
+
+        /**
+         * MainType: 229
+         * 229.001: Metering Value, values -2147483648..2147483647
+         */
+        dptMainTypeMap.put(229, DecimalType.class);
+        /** Exceptions Datapoint Types "4-Octet Signed Value", Main number 229 */
+        // Example: dptTypeMap.put(DptXlatorMeteringValue.DptMeteringValue.getID(), DecimalType.class);
+
+        /**
+         * MainType: 232
+         * 232.600: RGB, values 0 0 0 255 255 255 r g b
+         */
+        dptMainTypeMap.put(232, HSBType.class);
+        /** Exceptions Datapoint Types "RGB Color", Main number 232 */
+        // Example: dptTypeMap.put(DPTXlatorRGB.DPT_RGB.getID(), HSBType.class);
+
+        defaultDptMap = new HashMap<Class<? extends Type>, String>();
+        defaultDptMap.put(OnOffType.class, DPTXlatorBoolean.DPT_SWITCH.getID());
+        defaultDptMap.put(UpDownType.class, DPTXlatorBoolean.DPT_UPDOWN.getID());
+        defaultDptMap.put(StopMoveType.class, DPTXlatorBoolean.DPT_START.getID());
+        defaultDptMap.put(OpenClosedType.class, DPTXlatorBoolean.DPT_WINDOW_DOOR.getID());
+        defaultDptMap.put(IncreaseDecreaseType.class, DPTXlator3BitControlled.DPT_CONTROL_DIMMING.getID());
+        defaultDptMap.put(PercentType.class, DPTXlator8BitUnsigned.DPT_SCALING.getID());
+        defaultDptMap.put(DecimalType.class, DPTXlator2ByteFloat.DPT_TEMPERATURE.getID());
+        defaultDptMap.put(DateTimeType.class, DPTXlatorTime.DPT_TIMEOFDAY.getID());
+        defaultDptMap.put(StringType.class, DPTXlatorString.DPT_STRING_8859_1.getID());
+        defaultDptMap.put(HSBType.class, DPTXlatorRGB.DPT_RGB.getID());
+    }
+
+    /*
+     * (non-Javadoc)
+     *
+     * @see org.openhab.binding.knx1.config.KNXTypeMapper#toDPTValue(org.openhab.core.types.Type, java.lang.String)
+     */
+    @Override
+    public String toDPTValue(Type type, String dptID) {
+
+        DPT dpt;
+        int mainNumber = getMainNumber(dptID);
+        if (mainNumber == -1) {
+            LOGGER.error("toDPTValue couldn't identify mainnumber in dptID: {}", dptID);
+            return null;
+        }
+
+        try {
+            DPTXlator translator = TranslatorTypes.createTranslator(mainNumber, dptID);
+            dpt = translator.getType();
+        } catch (KNXException e) {
+            return null;
+        }
+
+        try {
+            // check for HSBType first, because it extends PercentType as well
```

Comment:
Maybe there are some things in this method that should better be split of

Reply:
This class is taken over as-is from OH1 (see e.g. openhab/openhab1-addons#5277). With all the weird s**t going on there, I wouldn't want to split the paths of both versions more than what is absolutely needed.